ompi/mca/coll/han/coll_han_dynamic_file.h

```diff
@@ -0,0 +1,110 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2020      The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2020      Bull S.A.S. All rights reserved.
+ *
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+#ifndef MCA_COLL_HAN_DYNAMIC_FILE_H
+#define MCA_COLL_HAN_DYNAMIC_FILE_H
+
+#include "ompi_config.h"
+
+#include "mpi.h"
+#include "ompi/mca/mca.h"
+#include "opal/util/output.h"
+
+/*
+ * ############################
+ * # MCA parameters interface #
+ * ############################
+ * An MCA parameter defined rule allows the user to choose which collective
+ * module will be used for a specific collective communication on a specific
+ * topological level. The standard name for these parameters is:
+ * [collective]_dynamic_[topologic_level]_module
+ *
+ * #######################
+ * # Dynamic file format #
+ * #######################
+ * File defined rules precede MCA parameter defined rule.
+ * To activate file reader, the MCA parameter use_dynamic_file_rules must
+ * be set to true. The path to the dynamic file is given by the MCA
+ * parameter dynamic_rules_filename. If there is any issue reading the file,
+ * the file is considered as invalid and only MCA parameter defined rules are
+ * used. If a potential logical issue is identified in the file, a
+ * warning is printed but the file is not considered as invalid.
+ *
+ * The file is built recursively.
+ * A set of rules of a type is built as follows:
+ * Number of rules of the set
+ * Rule1
+ * Rule2
+ * ...
+ *
+ * A rule of the level i is built as follows (excluding message size rule):
+ * Rule property
+ * Set of rules of level i+1
+ *
+ * A message size rule is built as follows:
+ * Message_size Component
+ *
+ * Rule properties are (by increasing level):
+ *     - Collective identifier:
+ *           Defined in ompi/mca/coll/base/coll_base_functions.h.
```

Comment:
I cannot identify the definitions in the mentioned header. Please provide a more detailed location

Reply:
COLLTYPE_T